File: 3.java
import java.util.Scanner;

public class Main {

    public static String isValidTime(String num) {
        int length = num.length();

        for (int i = 1; i < length; i++) {
            try {
                int hours = Integer.parseInt(num.substring(0, i));
                int minutes = Integer.parseInt(num.substring(i));
                if (hours >= 0 && hours <= 23 && minutes >= 0 && minutes <= 59) {
                    return "YES"; // корректное время найдено
                }
            } catch (NumberFormatException e) {
                return "NO"; 
            }
        }

        return "NO"; 
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("");
        String line = scanner.nextLine();
        String[] numbers = line.trim().split("\\s+");

        StringBuilder result = new StringBuilder();
        for (String num : numbers) {
            result.append(isValidTime(num)).append(" ");
        }

        System.out.println("");
        System.out.println(result.toString().trim());

        scanner.close(); 
    }
}

File: 3.swift
import Foundation

func isValidTime(_ num: String) -> String {
    let length = num.count
    for i in 1..<length {
        let index = num.index(num.startIndex, offsetBy: i)
        let hoursStr = String(num[..<index])
        let minutesStr = String(num[index...])

        if let hours = Int(hoursStr), let minutes = Int(minutesStr) {
            if hours >= 0 && hours <= 23 && minutes >= 0 && minutes <= 59 {
                return "YES"
            }
        }
    }
    return "NO"
}

if let input = readLine() {
    let numbers = input.split(separator: " ")
    var results: [String] = []
    for num in numbers {
        results.append(isValidTime(String(num)))
    }

    print(results.joined(separator: " "))
} else {
    print("Ошибка ввода.")
}
